### PR TITLE
Check that an identity is tied to a single profile.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by akatsoulas** [comment]
@johngian r?

File: mozillians/phonebook/views.py [hunk 1]
```diff
@@ -661,19 +661,25 @@ def get(self, request):
...
 
-            # Save the new identity to the IdpProfile
             user_q = {
-                'profile': profile,
                 'auth0_user_id': user_info['sub'],
                 'email': user_info['email']
             }
 
+            # Check that the identity doesn't exist in another profile
+            if IdpProfile.objects.filter(**user_q).exists():
```
Comment (johngian) [inline mozillians/phonebook/views.py]: This needs to be an OR not an AND. We want:

* The email to not exist to another IdpProfile

or

* The user_id to not exist to another IdpProfile

**Comment by johngian** [comment]
r+wc

File: mozillians/phonebook/views.py (hunk 1 above)
Comment (akatsoulas) [reply mozillians/phonebook/views.py]: this email is the one returned by Auth0 and not the one in the User object. It's not possible to have a combination of the same auth0_id and email in multiple identities. 

On the other hand is totally valid to have the same email in multiple Auth0 IDs. Eg a user that has the Google email used in GitHub.

Comment (akatsoulas) [reply mozillians/phonebook/views.py]: That said, if the above query returns True, then it means that a user already had added those. We don't care which profile in mozillians holds them. We just want those to be unique in the DB.

**Pull request merged**